processing/src/main/java/io/druid/segment/DimensionSelector.java

```diff
@@ -118,53 +118,48 @@
   @Nullable
   IdLookup idLookup();
 
-  /**
-   * @deprecated always throws {@link UnsupportedOperationException}
-   */
   @Deprecated
   @Override
   default float getFloat()
   {
-    throw new UnsupportedOperationException("DimensionSelector cannot be operated as numeric ColumnValueSelector");
+    // This is controversial, see https://github.com/druid-io/druid/issues/4888
+    return 0.0f;
```

Comment:
i thought we were gonna try and parse if possible.

Reply:
I preserved current behaviour. We can try to parse in a separate PR. Actually I tried to change this to parsing and it broke existing tests in `SchemaEvolutionTest`

Reply:
sounds good then